full-service/src/db/account.rs

```diff
@@ -585,6 +579,19 @@ impl AccountModel for Account {
 
         Ok(distinct_token_ids)
     }
+
+    fn next_subaddress_index(self, conn: &Conn) -> Result<u64, WalletDbError> {
+        use crate::db::schema::assigned_subaddresses;
+
+        let highest_subaddress_index: i64 = assigned_subaddresses::table
+            .filter(assigned_subaddresses::account_id.eq(&self.id))
+            .order_by(assigned_subaddresses::subaddress_index.desc())
+            .select(diesel::dsl::max(assigned_subaddresses::subaddress_index))
```

Comment:
do you need both of these select statements here?

Reply:
yeah, otherwise it returns the whole object which I don't want. It's not supposed to work that way, but alas, diesel.